client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -1,34 +1,50 @@
 import { useColorPaletteVariations, useFontPairingVariations } from '@automattic/global-styles';
 import { keyBy } from '@automattic/js-utils';
-import { useDispatch, useSelect } from '@wordpress/data';
-import { useState, useEffect, useRef, useCallback } from 'react';
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect, useRef } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { ONBOARD_STORE } from '../../../../../stores';
 import { decodePatternId } from '../utils';
 import type { Pattern, Category } from '../types';
 import type { OnboardSelect } from '@automattic/data-stores';
-import type { Design } from '@automattic/design-picker/src/types';
 import type { GlobalStylesObject } from '@automattic/global-styles';
 
 const useRecipe = ( siteId = 0, patterns: Pattern[], categories: Category[] ) => {
-	const incrementIndexRef = useRef( 0 );
-	const [ header, setHeader ] = useState< Pattern | null >( null );
-	const [ footer, setFooter ] = useState< Pattern | null >( null );
-	const [ sections, setSections ] = useState< Pattern[] >( [] );
-	const [ colorVariation, setColorVariation ] = useState< GlobalStylesObject | null >( null );
-	const [ fontVariation, setFontVariation ] = useState< GlobalStylesObject | null >( null );
+	const [ searchParams, setSearchParams ] = useSearchParams();
+
+	const header_pattern_id = searchParams.get( 'header_pattern_id' );
+	const footer_pattern_id = searchParams.get( 'footer_pattern_id' );
+
+	const section_pattern_ids = ( searchParams.get( 'pattern_ids' ) || '' )
+		.split( ',' )
+		.filter( Boolean );
+
+	const color_variation_title = searchParams.get( 'color_variation_title' );
+	const font_variation_title = searchParams.get( 'font_variation_title' );
+
+	const patternsById = keyBy( patterns, 'ID' );
+	const categoriesByName = keyBy( categories, 'name' );
+
+	const header = header_pattern_id ? patternsById[ decodePatternId( header_pattern_id ) ] : null;
+	const footer = footer_pattern_id ? patternsById[ decodePatternId( footer_pattern_id ) ] : null;
+
+	const sections = section_pattern_ids
```

Comment:
Whenever the `section_pattern_ids` changes, we will re-calculate the sections. However, we don't use the result after the component mounts. And there is a hidden logic that initialization of the keyed sections depends on the patterns and categories so we can init the keyed sections correctly when both patterns and categories are ready. Maybe we have to think about how to deal with it better

Reply:
Resolved via https://github.com/Automattic/wp-calypso/pull/80929/commits/7b514087743efe43ff916e750b603a81c3097aeb